thunder/executors/functional_teex.py

```diff
@@ -0,0 +1,236 @@
+    if bias is not None:
+        return False
+    return _linear_checker(a, w, bias)
+
+
+# Assuming one recipe per machine since it's based on hw capabilities.
+functional_te_recipe = get_default_fp8_recipe()
+
+
+def _te_fp8_recipe_meta(recipe_name: str):
+    return AnyProxy(recipe_name, prefix="r")
+
+
+def _te_fp8_recipe_impl(recipe_name: str):
+    return functional_te_recipe
+
+
+_te_fp8_recipe = functional_te_ex.register_operator("te_fp8_recipe", meta=_te_fp8_recipe_meta, fn=_te_fp8_recipe_impl)
+
+
+def _te_fp8_quantizers_meta(recipe_state: RecipeState, num_quantizers: int):
+    # giving num quantizers so there is no need to allocate extra things
+    return (*(AnyProxy(num_quantizers, prefix="q") for _ in range(num_quantizers)),)
+
+
+class TEQuantizerState:
+    def __init__(self):
+        self.quantizers = None
+
+    def __call__(self, recipe_state: RecipeState, num_quantizers: int):
+        if self.quantizers:
+            return self.quantizers
+        quantizers = recipe_state.make_quantizers()
+
+        self.quantizers = quantizers
+
+        return quantizers
+
+
+_te_fp8_quantizers = functional_te_ex.register_stateful_operator(
+    "te_fp8_quantizers", TEQuantizerState, meta=_te_fp8_quantizers_meta
+)
+
+
+def _te_fp8_state_meta(recipe, mode: str, num_quantizers: int, /):
+    # mode just to give the proxy something wihout allocating extra stuff
+    return AnyProxy(mode, prefix="s")
+
+
+class TERecipeState:
+    def __init__(self):
+        self.state = None
+
+    def __call__(self, recipe: Recipe, mode, num_quantizers):
+        if self.state:
+            return self.state
+
+        # mode is needed to get the correct dtypes inside for computation(setup quantizers)
+        recipe_state = te.fp8.RecipeState.create(
+            recipe,
+            mode=mode,
+            num_quantizers=num_quantizers,
+        )
+
+        self.state = recipe_state
+
+        return recipe_state
+
+
+_te_fp8_state = functional_te_ex.register_stateful_operator("te_fp8_state", TERecipeState, meta=_te_fp8_state_meta)
+
+
+def _linear_fwd_meta(
+    a: TensorProxy,
+    w: TensorProxy,
+    bias: TensorProxy | None,
+    input_quantizer: Float8Quantizer,
+    weight_quantizer: Float8Quantizer,
+):
+    return TensorProxy(like=a), TensorProxy(like=a), TensorProxy(like=w)
+
+
+def _linear_fwd_impl(a, w, bias, input_quantizer, weight_quantizer):
+    out, gemm_a, gemm_w = BasicLinear._functional_forward(
+        input=a,
+        weight=w,
+        with_quantized_compute=True,
+        input_quantizer=input_quantizer,
+        weight_quantizer=weight_quantizer,
+        output_quantizer=None,  # return out in original dtype (w.dtype)
+    )
+    return out, gemm_a, gemm_w
+
+
+_te_linear_fwd = functional_te_ex.register_operator(
+    "te_functional_linear_fwd", meta=_linear_fwd_meta, fn=_linear_fwd_impl
+)
+
+
+def _te_linear_execution_transform(a, w, /, bias):
+    recipe = _te_fp8_recipe("delayed")
```

Comment:
On Blackwell, default recipe is `MXFP8BlockScaling`. So the trace would say `delayed` but we would be using `MXFP8BlockScaling` under the hood.